python/src/sdf/pyElement.cc

```diff
@@ -0,0 +1,285 @@
+               "Set the parent of this Element.")
+          .def("set_name", &sdf::Element::SetName,
+               "Set the name of the Element")
+          .def("get_name", &sdf::Element::GetName, "Get the Element's name.")
+          .def("set_required", &sdf::Element::SetRequired,
+               "Set the requirement type.")
+          .def("get_required", &sdf::Element::GetRequired,
+               "Get the requirement string.")
+          // print_description
+          .def("set_explicitly_set_in_file",
+               &sdf::Element::SetExplicitlySetInFile,
+               "Set if the element and children where set or default in the "
+               "original file")
+          .def("get_explicitly_set_in_file",
+               &sdf::Element::GetExplicitlySetInFile,
+               "Return if the element was been explicitly set in the file")
+          // to_string
+          .def(
+              "add_attribute",
+              [](sdf::Element &_self, const std::string &_key,
+                 const std::string &_type, const std::string &_defaultvalue,
+                 bool _required, const std::string &_description = "")
+              {
+                sdf::Errors errors;
+                _self.AddAttribute(_key, _type, _defaultvalue, _required,
+                                   errors, _description);
+                ThrowIfErrors(errors);
+              },
+              "Add an attribute value.")
+          .def(
+              "add_value",
+              [](sdf::Element &_self, const std::string &_type,
+                 const std::string &_defaultValue, bool _required,
+                 const std::string &_description = "")
+              {
+                sdf::Errors errors;
+                _self.AddValue(_type, _defaultValue, _required, errors,
+                               _description);
+                ThrowIfErrors(errors);
+              },
+              "Add a value to this Element")
+          .def(
+              "add_value",
+              [](sdf::Element &_self, const std::string &_type,
+                 const std::string &_defaultValue, bool _required,
+                 const std::string &_minValue, const std::string &_maxValue,
+                 const std::string &_description = "")
+              {
+                sdf::Errors errors;
+                _self.AddValue(_type, _defaultValue, _required, _minValue,
+                               _maxValue, errors, _description);
+                ThrowIfErrors(errors);
+              },
+              "Add a value to this Element")
+          // get_attribute (string)
+          .def("get_attribute_count", &sdf::Element::GetAttributeCount,
+               "Get the number of attributes.")
+          // get_attributes
+          // get_attribute (index)
+          // get_element_description_count
+          // get_element_description (index)
+          // get_element_description (string)
+          // has_element_description
```

Comment:
are these TODOs ?

Reply:
Yes, I've implemented some of them. The rest, I have commented why I didn't think we need to implement them.